server/src/test/kotlin/fi/oph/ludos/assignment/AssignmentControllerTest.kt

```diff
@@ -373,4 +392,48 @@ class AssignmentControllerTest(@Autowired val mockMvc: MockMvc) {
     fun getAssignmentsWithNoRole() {
         mockMvc.perform(getAllAssignments(Exam.SUKO)).andExpect(status().is3xxRedirection())
     }
+
+    @Test
+    @WithOpettajaRole
+    fun getAssignmentDraftAsOpettaja() {
+        idsOfAssignmentDrafts.forEach() {
+            mockMvc.perform(getAssignment(Exam.SUKO, it)).andExpect(status().isNotFound())
+        }
+    }
+
+    @Test
+    @WithOpettajaRole
+    fun getAssignmentsAsOpettaja() {
+        val res = mockMvc.perform(getAllAssignments(Exam.SUKO)).andExpect(status().isOk())
+            .andReturn().response.contentAsString
+
+        val assignments = objectMapper.readValue(res, Array<TestAssignmentSukoOut>::class.java)
+
+        assertTrue(
+            assignments.none { it.publishState == PublishState.DRAFT }, "Opettaja should not see draft assignments"
+        )
+    }
+
+    @Test
+    @WithOpettajaRole
+    fun assignmentTestInsufficientRole() {
+        val testAssignmentStr = """{
+            "exam": "SUKO",
+            "nameFi": "suomi",
+            "nameSv": "ruotsi",
+            "contentFi": "suomi",
+            "contentSv": "ruotsi",
+            "instructionFi": "suomi",
+            "instructionSv": "ruotsi",
+            "publishState": "PUBLISHED",
+            "assignmentTypeKoodiArvo": "003",
+            "oppimaaraKoodiArvo": "ET",
+            "tavoitetasoKoodiArvo": "0004",
+            "aiheKoodiArvos": ["002", "003"],
+            "laajaalainenOsaaminenKoodiArvos": ["06", "03"]
+        }""".trimIndent()
```

Comment:
Ihan vaan huomiona, ei vaadi toimenpiteitä tässä vaiheessa:

Näitä string-muotosia payloadeja kandeis välttää, koska menee hankalaks, jos vaik lisätään uus kenttä tjsp. Mut tässä tapauksessa payloadilla ei taida olla kauheasti väliä, kun ton pitäis posahtaa PreAuthorizeen toivottavasti ennen kuin payloadia yritetään edes parsia.